core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -345,7 +344,9 @@ public BlockOutStream getOutStream(long blockId, long blockSize, OutStreamOption
         break;
       }
       workerAddressList.add(address);
-      blockWorkers.removeAll(blockWorkersByHost.get(address.getHost()));
+      List<BlockWorkerInfo> updatedInfos = Lists.newArrayList(workerOptions.getBlockWorkerInfos());
```

Comment:
Instead of making a copy of the current set of `BlockWorkerInfos` here could we utilize a single reference to a set?

Reply:
Also `Lists.newArrayList` [shouldn't be used](https://google.github.io/guava/releases/23.0/api/docs/com/google/common/collect/Lists.html#newArrayList--) any more

> Note for Java 7 and later: this method is now unnecessary and should be treated as deprecated. Instead, use the ArrayList constructor directly, taking advantage of the new "diamond" syntax.

Reply:
`getBlockWorkerInfos` returns an iterable and not a collection so we cannot use the ArrayList constructor.

Reply:
I updated it to use a single copied list. I still copied the list once since I think this is a special case and wanted to make it explicit.

Reply:
Cool, thanks!